pages/api/send-analytics.ts

```diff
@@ -0,0 +1,32 @@
+import getConfig from 'next/config';
+import moment from 'moment/moment';
+import {Analytics} from '@segment/analytics-node'
+import { NextApiRequest, NextApiResponse } from 'next';
+
+const { serverRuntimeConfig } = getConfig();
+
+export default async function sendAnalytics(req: NextApiRequest, res: NextApiResponse) {
+    const {event, data = {}} = req?.body;
+    const anonymousId = req.cookies['helio_analytics_id'] || moment().unix().toString()
+    const signedUserId = req.cookies['user_id']
+
+    try {
+        const client = new Analytics({
+            writeKey: serverRuntimeConfig.SEGMENT_WRITE_KEY,
+        })
+    
+        client.track({
+            event,
+            anonymousId,
+            properties: {
+                ...data,
+                signedUserId,
+            }
+        })
+    } catch (e) {
+        console.log("Error sending analytics", e)
+    } finally {
+        res.setHeader('Set-Cookie', 'helio_analytics_id=' + anonymousId + '; Path=/; HttpOnly; SameSite=Strict; Max-Age=' + 60 * 60 * 24 * 30)
+        return res.json({success: true})
+    }
+}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `sendAnalytics` function is not handling errors properly. If an error occurs when sending analytics data, it simply logs the error and continues execution as if nothing happened. This could lead to misleading results because the client might receive a success response even though the analytics data was not sent successfully.

Instead of just logging the error, you should return an error response to the client. This way, the client can handle the error appropriately (e.g., by retrying the request or showing an error message to the user).

Here's how you can modify the code to return an error response:

```diff
  try {
      const client = new Analytics({
          writeKey: serverRuntimeConfig.SEGMENT_WRITE_KEY,
      })
  
      client.track({
          event,
          anonymousId,
          properties: {
              ...data,
              signedUserId,
          }
      })
- } catch (e) {
-     console.log("Error sending analytics", e)
+ } catch (error) {
+     console.error("Error sending analytics", error);
+     return res.status(500).json({ success: false, message: 'Failed to send analytics data.' });
  } finally {
      res.setHeader('Set-Cookie', 'helio_analytics_id=' + anonymousId + '; Path=/; HttpOnly; SameSite=Strict; Max-Age=' + 60 * 60 * 24 * 30)
      return res.json({success: true})
  }
```

Also, consider using a more secure method for generating the `anonymousId`. Using the current Unix timestamp as the ID could potentially lead to collisions if multiple requests are made at the same time. Consider using a UUID or another method that guarantees uniqueness.

Lastly, the cookie being set in the `finally` block will always overwrite any existing 'helio_analytics_id' cookie, regardless of whether the tracking was successful or not. This could potentially lead to loss of tracking data if the same user makes multiple requests. Consider setting the cookie only when a new `anonymousId` is generated.

```diff
- const anonymousId = req.cookies['helio_analytics_id'] || moment().unix().toString()
+ let anonymousId = req.cookies['helio_analytics_id'];
+ if (!anonymousId) {
+     anonymousId = uuidv4(); // assuming you've imported uuid/v4 as uuidv4
+     res.setHeader('Set-Cookie', 'helio_analytics_id=' + anonymousId + '; Path=/; HttpOnly; SameSite=Strict; Max-Age=' + 60 * 60 * 24 * 30);
+ }
```
In this case, you would need to import the `uuid` library at the top of your file:
```javascript
import { v4 as uuidv4 } from 'uuid';
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->